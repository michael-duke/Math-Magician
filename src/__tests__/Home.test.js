const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Home = require('../pages/Home').default;

describe('Tests for Home Component', () => {
  it('Renders Home page', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  it("Check if heading 'Welcome to our page' exists", () => {
    render(<Home />);
    const welcomeHeading = screen.getByRole('heading', { name: /Welcome to our page!/i })
    expect(welcomeHeading).toBeInTheDocument();
  });
});
