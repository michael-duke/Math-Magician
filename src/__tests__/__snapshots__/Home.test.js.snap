// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home Component Renders Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="home mt-7 p-12 space-y-10"
      >
        <h1
          class="font-semibold text-2xl"
        >
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="home mt-7 p-12 space-y-10"
    >
      <h1
        class="font-semibold text-2xl"
      >
        Welcome to our page!
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
